src/components/Email/ManageWalletModalWithEmail.tsx

```diff
@@ -0,0 +1,193 @@
+import { useCallback, useEffect, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import breakpoints from '~/components/core/breakpoints';
+import { Button } from '~/components/core/Button/Button';
+import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { BaseM, TitleDiatypeL, TitleDiatypeM } from '~/components/core/Text/Text';
+import Toggle from '~/components/core/Toggle/Toggle';
+import EmailManager from '~/components/Email/EmailManager';
+import ManageWallets from '~/components/ManageWallets/ManageWallets';
+import { useReportError } from '~/contexts/errorReporting/ErrorReportingContext';
+import { useToastActions } from '~/contexts/toast/ToastContext';
+import { ManageWalletModalWithEmailFragment$key } from '~/generated/ManageWalletModalWithEmailFragment.graphql';
+
+import useUpdateEmailNotificationSettings from './useUpdateEmailNotificationSettings';
+
+type Props = {
+  queryRef: ManageWalletModalWithEmailFragment$key;
+  newAddress?: string;
+  onEthAddWalletSuccess?: () => void;
+  onTezosAddWalletSuccess?: () => void;
+};
+
+function ManageWalletsModalWithEmail({
+  newAddress,
+  queryRef,
+  onEthAddWalletSuccess,
+  onTezosAddWalletSuccess,
+}: Props) {
+  const query = useFragment(
+    graphql`
+      fragment ManageWalletModalWithEmailFragment on Query {
+        viewer @required(action: THROW) {
+          ... on Viewer {
+            email @required(action: THROW) {
+              email
+              emailNotificationSettings {
+                unsubscribedFromAll
+                unsubscribedFromNotifications
+              }
+            }
+          }
+        }
+
+        ...EmailManagerFragment
+        ...ManageWalletsFragment
+      }
+    `,
+    queryRef
+  );
+
+  const updateEmailNotificationSettings = useUpdateEmailNotificationSettings();
+  const [isEmailNotificationChecked, setIsEmailNotificationChecked] = useState(false);
+  const [isShowAddEmail, setIsShowAddEmail] = useState(false);
+  const { pushToast } = useToastActions();
+  const reportError = useReportError();
+
+  // Invert the value since in the db stored as unsubscribe while in the UI it's subscribe
+  const isEmailNotificationSubscribed =
+    query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromNotifications || false;
+
+  const isEmailUnsubscribedFromAll =
+    !!query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromAll ?? false;
+
+  // Set the initial state of the email notification toggle
+  // so we can instantly toggle it on/off without waiting for the mutation to complete
+  useEffect(() => {
+    setIsEmailNotificationChecked(isEmailNotificationSubscribed);
+  }, [isEmailNotificationSubscribed]);
+
+  // If the user already have email attached, toggle the email manager ui
+  const userEmail = query?.viewer?.email?.email;
+  useEffect(() => {
+    setIsShowAddEmail(Boolean(userEmail));
+  }, [userEmail]);
```

Comment:
same thing here

Reply:
I want to call out that the tradeoff here is that if the Relay store updates, our local state won't update anymore. But I feel strongly that you wouldn't want some change in the Relay store to cause "random" changes to happen to a form you're submitting.

Reply:
if that is what we end up wanting, that's fine, but if we initialize `useState` appropriately, at least we won't have a double render.

Reply:
TIL: Didn't know that the relay store affect the local state too. 

I've removed this `useEffect` and set the response as a initial local state.